core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2287,6 +2287,23 @@ private[spark] object Utils extends Logging with SparkClassUtils {
     }.map(threadInfoToThreadStackTrace)
   }
 
+  /** Return a heap dump. Used to capture dumps for the web UI */
+  def getHeapHistogram(): Array[String] = {
+    // From Java 9+, we can use 'ProcessHandle.current().pid()'
+    val pid = getProcessName().split("@").head
+    val builder = new ProcessBuilder("jmap", "-histo:live", pid)
+    builder.redirectErrorStream(true)
```

Comment:
Log errors in invocation to executor logs instead of sending it to driver as response ?

Reply:
That's totally different feature. This is designed as a part of Spark Driver UI, @mridulm .